canary.go

```diff
@@ -37,6 +38,29 @@ func main() {
 		log.Fatalf("error reading file: %v\n", e)
 	}
 
+	pid := os.Getpid()
+	pidFileName := "/run/canary.pid"
+	_, err = os.Stat(pidFileName)
+	if !os.IsNotExist(err) {
+
+		pidFile, e := ioutil.ReadFile(pidFileName)
+		if e != nil {
+			log.Fatalf("error reading PID file: %v\n", e)
+		}
+
+		if string(pidFile) != strconv.Itoa(pid) {
+			cmd := exec.Command("kill", "-9", string(pidFile))
+			stdoutStderr, err := cmd.CombinedOutput()
+			log.Printf("%s\n", stdoutStderr)
+			if err == nil {
+				log.Printf("killed previous process with PID %s", string(pidFile))
+			}
+
+		}
+	}
+
+	err = ioutil.WriteFile(pidFileName, []byte(strconv.Itoa(pid)), 0644)
```

Comment:
I don't think this is quite the right thing, having a process run kill -9 is pretty terrifying.

Instead we should add a context with a deadline to all our calls to guarantee that none of these take more than a minute to run, which should take care of making sure there aren't phantom process sticking around.